File: ts/impots-entreprise-kata/src/domain/auto.entreprise.ts
import {Entreprise} from "./entreprise";
import {TypeEntreprise} from "./entreprise.type";

export class AutoEntreprise extends Entreprise {

    constructor(siret: string, denomination: string) {
        super(siret, denomination, TypeEntreprise.AUTOENTREPRISE);
    }

    tauxImposition(): number {
        return 0.25;
    }

}

File: java/impots-entreprise-kata/src/main/java/fr/soat/cleancoders/domain/AutoEntreprise.java
package fr.soat.cleancoders.domain;

public class AutoEntreprise extends Entreprise {

    public AutoEntreprise(String siret, String denomination) {
        super(siret, denomination, TypeEntreprise.AUTOENTREPRISE);
    }

    @Override
    public Double tauxImposition() {

        throw new UnsupportedOperationException("NOT YET IMPLEMENTED");

    }

}
